### PR TITLE
ConditionCountMetric fix and docstrings

### PR DESCRIPTION
**Pull request by richard-rogers**
## Description

Adds docstrings & fix failed condition logging & exception bug.

- [ ] I have reviewed the [Guidelines for Contributing](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md).

### CODE REVIEW

File: python/whylogs/core/metrics/condition_count_metric.py [hunk 1]
```diff
@@ -147,12 +211,15 @@ def columnar_update(self, data: PreprocessedColumn) -> OperationResult:
...
 
         self.total.set(self.total.value + count)
-        if failed_conditions:
+        for failed in failed_conditions:
+            condition = self.conditions[failed]
             if condition.log_on_failure:
-                logger.warning(f"Condition(s) {', '.join(failed_conditions)} failed")
+                logger.warning(f"Condition {failed} failed")
 
+        for failed in failed_conditions:
```
Comment (jamie256) [inline python/whylogs/core/metrics/condition_count_metric.py]: Can you explain why this is correct and what the bug was before? It looks like previously the error raised would contain a message for each failed condition in the set of failed conditions on that column's data, but now the error message would reference the first failed condition?

Comment (richard-rogers) [reply python/whylogs/core/metrics/condition_count_metric.py]: The bug was that the `condition` used in the logging or throwing code was previously the last condition evaluated in the loop on line 136/200, whether it was the among the failed conditions or not.  I can restore the behavior of including all the failed throwing conditions if required.

Comment (jamie256) [reply python/whylogs/core/metrics/condition_count_metric.py]: I think we should preserve returning the set of failures.

I don't follow that bug description, it looks like the failed_conditions is a set of the failures isn't it in the original near line 152:

`logger.warning(f"Condition(s) {', '.join(failed_conditions)} failed")`

Comment (richard-rogers) [reply python/whylogs/core/metrics/condition_count_metric.py]: It was a scope bug... the `condition` tested in the `if`s for logging and throwing was the `condition` from the last iteration of the evaluation loop, whether that condition was configured to log or throw (or even if it matched).

I restored the `join`s on the offending condition names, though segregated for those that should log and those that should throw.

Comment (jamie256) [reply python/whylogs/core/metrics/condition_count_metric.py]: great! ok, I see its the throws/not throws wasn't respecting the per condition setting. I thought you were claiming it ignored the failure of condition checks on earlier conditions (when they are all set to throw on error).

**Review by jamie256 — APPROVED** [review]
LGTM!

**Pull request merged**